src/sentry/snuba/metrics/naming_layer/mri.py

```diff
@@ -264,9 +264,12 @@ def format_mri_field(field: str) -> str:
 
         if parsed:
             if condition_id := parsed.mri.span_attribute_rule_id:
-                condition = SpanAttributeExtractionRuleCondition.objects.get(id=condition_id)
-                config = condition.config
-                return f'{parsed.op}({config.span_attribute}) filtered by "{condition.value}"'
+                try:
+                    condition = SpanAttributeExtractionRuleCondition.objects.get(id=condition_id)
+                    config = condition.config
+                    return f'{parsed.op}({config.span_attribute}) filtered by "{condition.value}"'
+                except SpanAttributeExtractionRuleCondition.DoesNotExist:
+                    return field
```

Comment:
Can you also add a metric or some logging, because, if I am not mistaken, this situation shouldn't be happening?

Reply:
Yeah, at the moment it can still work because you can delete a metric without it deleting an alert. (when https://github.com/getsentry/sentry/issues/74255 is done this problem goes away)